parsers/robot.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * robot.c
+ * parser for robot framewor
+ * Author - Daniel Riechers
+ * License GPL-2
+ */
+
+#include "general.h"
+
+#include <string.h>
+
+#include "parse.h"
+#include "vstring.h"
+#include "routines.h"
+
+typedef enum {
+        K_KEYWORD,
+        K_VARIABLE,
+        COUNT_KIND
+} RobotKind;
+
+static kindOption RobotKinds[COUNT_KIND] = {
+	{true, 'k', "keyword",    "keywords"},
+	{true, 'v', "variable",   "variables"},
+};
+
+static void findRobotTags (void)
+{
+	findRegexTags ();
+}
+
+static void whitespaceSwap (vString *const s)
+{
+        char replaceWith = '_';
+        char toReplace = ' ';
+
+        if(strchr(s->buffer, '_'))
+        {
+            replaceWith = ' ';
+            toReplace = '_';
+        }
+
+        for(int i=0; i < s->length; i++)
+            if(s->buffer[i] == toReplace)
+                s->buffer[i] = replaceWith;
+}
+
+static void keywords (const char *const line, const regexMatch *const matches,
+                               const unsigned int count, void *data)
+{
+    if (count > 1)
+    {
+        vString *const name = vStringNew ();
+        vStringNCopyS (name, line + matches [1].start, matches [1].length);
+        makeSimpleTag (name, RobotKinds, K_KEYWORD);
+        whitespaceSwap(name);
+        makeSimpleTag (name, RobotKinds, K_KEYWORD);
```

Comment:
White space swap doesn't have to do with python keywords. White space swap is there because in robot spaces are the same as underscores except in the case of delimiting white space which is two or more spaces. Python keywords can in fact be called with spaces or underscores.

`Keyword ABC 123
    [Arguments]      ${arg1}      ${arg2}
    foo   ${arg1}
    bar    ${arg2}
`

The calls to the keyword above are equivalent:
`Keyword ABC 123      this is my arg1      this is my arg2` or `Keyword_ABC_123      this is my arg1      this is my arg2`